group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
```

Comment:
I haven't done any refactoring yet, only described what I have found. Basically, it looks like "Activity" is an instance of Group Project block (`GroupActivity`) and "Stage" is `ActivityComponent`. Hence, sections are parts of stage, and all `ActivitySections` within `ActivityComponent` are shown simultaneously.

Looks like the idea behind this was to allow special sections, like "uploads" and "resources" - they do not require anything but their type and automatically populated with content - along with custom sections containing (almost) arbitrary HTML. Since we're to move uploads and resources from activity content to Project Navigator, the need for sections might disappear.